x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/search/SparseVectorQueryBuilder.java

```diff
@@ -332,6 +350,49 @@ public TransportVersion getMinimalSupportedVersion() {
         return TransportVersions.V_8_15_0;
     }
 
+    private TokenPruningConfig getTokenPruningConfigForQuery(MappedFieldType ft, SearchExecutionContext context) {
+        // if we do not have searcher, there can be no token pruning because there will be no IndexReader
+        if (context.searcher() == null) {
+            return null;
+        }
+
+        // if we are not on a supported index version, do not prune by default
+        // nor do we check the index options, so, we'll return a pruning config only if the query specifies it.
+        if (context.indexVersionCreated().onOrAfter(IndexVersions.SPARSE_VECTOR_PRUNING_INDEX_OPTIONS_SUPPORT) == false
+            && context.indexVersionCreated()
+                .between(
+                    IndexVersions.SPARSE_VECTOR_PRUNING_INDEX_OPTIONS_SUPPORT_BACKPORT_8_X,
+                    IndexVersions.UPGRADE_TO_LUCENE_10_0_0
+                ) == false) {
+            return (shouldPruneTokens != null && shouldPruneTokens) ? tokenPruningConfig : null;
+        }
+
+        Boolean shouldQueryPruneTokens = shouldPruneTokens;
+        TokenPruningConfig pruningConfigToUse = tokenPruningConfig;
+
+        if (ft instanceof SparseVectorFieldMapper.SparseVectorFieldType asSVFieldType && asSVFieldType.getIndexOptions() != null) {
+            shouldQueryPruneTokens = shouldQueryPruneTokens == null ? asSVFieldType.getIndexOptions().getPrune() : shouldQueryPruneTokens;
+            pruningConfigToUse = pruningConfigToUse == null ? asSVFieldType.getIndexOptions().getPruningConfig() : pruningConfigToUse;
+        }
+
+        // do not prune if shouldQueryPruneTokens is explicitly set to false
+        if (shouldQueryPruneTokens != null && shouldQueryPruneTokens == false) {
+            return null;
+        }
+
+        // if we're here, we should prune if set or by default
+        // if we don't have a pruning config, use the default
+        pruningConfigToUse = pruningConfigToUse == null
+            ? new TokenPruningConfig(
+                TokenPruningConfig.DEFAULT_TOKENS_FREQ_RATIO_THRESHOLD,
+                TokenPruningConfig.DEFAULT_TOKENS_WEIGHT_THRESHOLD,
+                false
+            )
+            : pruningConfigToUse;
+
+        return pruningConfigToUse;
```

Comment:
This logic should be in the mapper. Why aren't we defaulting specifically to `true` in the mapper?